src/cpp/session/modules/SessionReticulate.R

```diff
@@ -1411,3 +1462,403 @@ html.heading = _heading
    else
       item
 })
+
+#    > str(.rs.describeObject(globalenv(), "a"))
+# List of 10
+#  $ name             : 'rs.scalar' chr "a"
+#  $ type             : 'rs.scalar' chr "numeric"
+#  $ clazz            : chr [1:2] "numeric" "double"
+#  $ is_data          : 'rs.scalar' logi FALSE
+#  $ value            : 'rs.scalar' chr "1"
+#  $ description      : 'rs.scalar' chr " num 1"
+#  $ size             : 'rs.scalar' num 56
+#  $ length           : 'rs.scalar' int 1
+#  $ contents         : list()
+#  $ contents_deferred: 'rs.scalar' logi FALSE
+.rs.addFunction("reticulate.describeObject", function(name, parent)
+{
+   builtins <- reticulate::import_builtins(convert = TRUE)
+   object <- if (inherits(parent, "python.builtin.dict"))
+      reticulate::py_get_item(parent, name)
+   else
+      reticulate::py_get_attr(parent, name)
+   
+   # is this object 'data'? consider non-callable, non-module objects as data
+   isData <- !(
+      grepl("^__.*__$", name) ||
+      reticulate:::py_is_callable(object) ||
+      reticulate:::py_is_module(object)
+   )
+   
+   # TODO: there isn't really a distinction between an objects "type"
```

Comment:
I've converted this to a NOTE, since it really doesn't represent an actionable item -- it's ultimately just to help anyone reading that bit of code understand some of the decisions made there.